sys/linux/auto.txt

```diff
@@ -1059,63 +1039,271 @@ syz_genetlink_get_family_id$auto_tcp_metrics(name ptr[in, string["tcp_metrics"]]
+utimes$auto(filename ptr[inout, int8], t ptr[inout, old_timeval32$auto_record]) (automatic)
+vmsplice$auto(fd int32, uiov ptr[inout, iovec$auto_record], nr_segs int32, flags int32) (automatic)
+wait4$auto(upid int32, stat_addr ptr[inout, int32], options int32, ru ptr[inout, rusage$auto_record]) (automatic)
+waitid$auto(which int32, upid int32, infop ptr[inout, siginfo$auto_record], options int32, ru ptr[inout, rusage$auto_record]) (automatic)
+waitpid$auto(pid int32, stat_addr ptr[inout, int32], options int32) (automatic)
+write$auto(fd int32, buf ptr[inout, int8], count int32) (automatic)
+writev$auto(fd int32, vec ptr[inout, iovec$auto_record], vlen int32) (automatic)
 _ = __NR_mmap2
 
+XFRMA_COADDR$auto_record [
+	a4	int32
+	a6	array[int32, 4]
+	in6	in6_addr$auto_record
+]
+
+XFRMA_SRCADDR$auto_record [
+	a4	int32
+	a6	array[int32, 4]
+	in6	in6_addr$auto_record
+]
+
+__aio_sigset$auto_record {
+	sigmask		ptr[inout, sigmask$auto_record]
+	sigsetsize	int32
+}
+
+__kernel_itimerspec$auto_record {
+	it_interval	__kernel_timespec$auto_record
+	it_value	__kernel_timespec$auto_record
+}
+
+__kernel_old_itimerval$auto_record {
+	it_interval	__kernel_old_timeval$auto_record
+	it_value	__kernel_old_timeval$auto_record
+}
+
+__kernel_old_timeval$auto_record {
+	tv_sec	int32
+	tv_usec	int32
+}
+
+__kernel_sockaddr_storage$auto_record {
+	__kernel_sockaddr_storage_0	__kernel_sockaddr_storage_0$auto_record
+}
+
+__kernel_sockaddr_storage_0$auto_record [
+	__kernel_sockaddr_storage_0_0	__kernel_sockaddr_storage_0_0$auto_record
+	__align				ptr[inout, void]
+]
+
+__kernel_sockaddr_storage_0_0$auto_record {
+	ss_family	int16
+	__data		array[int8, 126]
+}
+
+__kernel_timespec$auto_record {
+	tv_sec	intptr
+	tv_nsec	intptr
+}
+
+__kernel_timex$auto_record {
+	modes			int32
+	__kernel_timex_1	int32
+	offset			intptr
+	freq			intptr
+	maxerror		intptr
+	esterror		intptr
+	status			int32
+	__kernel_timex_7	int32
+	constant		intptr
+	precision		intptr
+	tolerance		intptr
+	time			__kernel_timex_timeval$auto_record
+	tick			intptr
+	ppsfreq			intptr
+	jitter			intptr
+	shift			int32
+	__kernel_timex_16	int32
+	stabil			intptr
+	jitcnt			intptr
+	calcnt			intptr
+	errcnt			intptr
+	stbcnt			intptr
+	tai			int32
+	__kernel_timex_23	int32
+	__kernel_timex_24	int32
+	__kernel_timex_25	int32
+	__kernel_timex_26	int32
+	__kernel_timex_27	int32
+	__kernel_timex_28	int32
+	__kernel_timex_29	int32
+	__kernel_timex_30	int32
+	__kernel_timex_31	int32
+	__kernel_timex_32	int32
+	__kernel_timex_33	int32
+}
+
+__kernel_timex_timeval$auto_record {
+	tv_sec	intptr
+	tv_usec	intptr
+}
+
+__old_kernel_stat$auto_record {
+	st_dev		int16
+	st_ino		int16
+	st_mode		int16
+	st_nlink	int16
+	st_uid		int16
+	st_gid		int16
+	st_rdev		int16
+	st_size		int32
+	st_atime	int32
+	st_mtime	int32
+	st_ctime	int32
+}
+
+__sifields$auto_record [
+	_kill		_kill$auto_record
```

Comment:
Nit: I think it's cleaner to name anonymous structs like `union_name + _ + field_name`, e.g. `__sifields__kill` in this case. Just using the field name is too ambiguous.